### ISSUE TITLE
No longer able to extract tags

### ISSUE THREAD

**Issue by dromer** [robertmuth/apetag#21]
Previously one could simply `apetag -i $inputfile -file $outputfile` however this flag is no longer available.

The help output seems to ommit any method to write the apetag to disk, but is there still a method to do this?

We are currently pinning to the last known version to work which is `55f3d7a51946787df401573fc92698ed88a2cdee`

**Comment by smithjd15** [comment]
This use was expected to be more for debugging, and it's usefulness was questionable because one could apetag -i $outputfile -m overwrite -file $inputfile to apply inputfile's tag to outputfile.

From a quick google: https://unix.stackexchange.com/questions/231289/extract-data-between-two-matched-patterns-in-a-binary-file . The pattern to match is APETAGEX at both the beginning and end.

**Comment by robertmuth** [comment]
@dromer 
Since I forgot what "-file" did, could you add a little more context. Maybe rephrase this as a feature requests and describe your use case.

**Comment by dromer** [comment]
It allows one to write the extracted tags to a file.

**Comment by robertmuth** [comment]
do you have any preferences with respect to formatting? Would stdout work?
Do you expect the info to be fed back into apetag?

If you could describe your use case a bit more. I can make educated guesses and avoid this back and forth through the bug.

**Comment by dromer** [comment]
stdout would be fine I think. can then always redirect to a file. or other parser, etc.

Our use-case was the extraction of tags to be processed by a script later.
The code I was using for that is here -> https://git.idiopolis.nl/dreamer/PreviouslyOnIFM/src/branch/develop/splimport/extracttags.sh

**Comment by dromer** [comment]
Edit actually I realize that in our use-case the stdout approach may not work. These files contained binary data, so I'm not sure how one could get this over stdout.

(in our case this was complicated because the program that created the tags inserted some custom table inside the tag that we had to extract and parse separately)

**Comment by robertmuth** [comment]
I added a new mode that dumps the tags like so:

```
./apetag -m readasflags -i TestData/empty_ape.mp3 
-p one=id3v1magicdetecttag1
-p two=id3v1magicdetecttagtwo2
```

I still need to add escaping support for binary data.
AlternativeIy,  could just skip binary data.
Will this approach work for your use case?

**Comment by dromer** [comment]
Hehe, no because we explicitly need the binary data in our case. But honestly this was an older project and worst case we just use that specific pinned version of apetag in the future. (the binary data was also of different endian-ness and in some microsoft/dos format. [don't ask](https://git.idiopolis.nl/dreamer/PreviouslyOnIFM/src/branch/develop/splimport/decode.py#L53).)

Thnx for looking into it though! I think it's a useful feature to extract just the apetag from a file. and then be able to process that on its own. In our case we ended up writing a parser using Kaitai Struct so we could really lift the raw binary out of the apetag. It was easiest to then have _just_ the apetag (and no "preamble" of varying structures mediafile attached).

Hope it makes a bit more sense now. We really needed to extract just the apetag part of the file in this case.

**Comment by robertmuth** [comment]
Added a new mode  "readasflags" that will be used like so:

`apetag -i  [audio-file] -m readasflags -fileprefix tagfile.`

This will print something like:

```
  -p Test=File
  -f bin_tag=tagfile.bin_tag
```
tagfile.bin_tag contains the content of the binary tag [bin_tag]
